airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java

```diff
@@ -19,6 +20,20 @@ default ColumnId buildColumnId(final String name) {
 
   ColumnId buildColumnId(String name, String suffix);
 
+  /**
+   * Implementations which need to perform catalog-agnostic setup may override this method. For
+   * example, any {@code CREATE FUNCTION} calls should happen here.
+   * <p>
+   * It's expected that most implementations do not need to override this method.
+   * <p>
+   * This SQL MUST be idempotent and threadsafe. Callers SHOULD guarantee this method is only called
```

Comment:
if the sync lifecycle is container setup & teardown, see other comment about using an atomic var. will that give guarantees with in this object's lifecycle.